From 0cff0dbfc0813d01a0ebbd3910519bfac4d9eec2 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 26 Apr 2013 18:16:13 +0200
Subject: [PATCH 33/35] MINOR: init: indicate the SSL runtime version on -vv.

It happens that openssl's API can differ between versions, causing some
serious trouble if the version used at runtime is not the same as used
for building.

Now we report the two versions separately along with a warning if the
version differs (except the patch version).
---
 src/haproxy.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/haproxy.c b/src/haproxy.c
index da7a625..6155ea3 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -259,6 +259,9 @@ void display_build_opts()
 
 #ifdef USE_OPENSSL
 	printf("Built with OpenSSL version : " OPENSSL_VERSION_TEXT "\n");
+	printf("Running on OpenSSL version : %s%s\n",
+	       SSLeay_version(SSLEAY_VERSION),
+	       ((OPENSSL_VERSION_NUMBER ^ SSLeay()) >> 8) ? " (VERSIONS DIFFER!)" : "");
 	printf("OpenSSL library supports TLS extensions : "
 #if OPENSSL_VERSION_NUMBER < 0x00907000L
 	       "no (library version too old)"
-- 
1.8.1.5

